event-internal.h

```diff
@@ -42,6 +42,14 @@ extern "C" {
 #include "mm-internal.h"
 #include "defer-internal.h"
 
+#ifndef TAILQ_FOREACH_SAFE
+#define TAILQ_FOREACH_SAFE(var, head, field, tvar)        \
+    for ((var) = TAILQ_FIRST((head));                     \
+         (var) && ((tvar) = TAILQ_NEXT((var), field), 1); \
```

Comment:
will this gcc'ism works for win32?

Reply:
Been using it for 8 years now in evhtp, never ran into an issue.

Not sure how this is a gcc'ism; it just uses `tmp` to hold onto the next element, so `var` can be freed, or removed inline.

Reply:
Egh, confused `(foo = 1, 1)` in a `for` loop condition with `({ compound statement })`